1 коммит

ebatuev opened the pull request.

Commit:
1 коммит

egorvas commented on src/main/java/pages/MainPage.java:
```diff
@@ -0,0 +1,93 @@
+
+    /** Объявил методы клика по каждому вопросу в «Вопросы о важном» **/
+    // методы клика по «Вопросы о важном»
+    public void openFAQ1() { driver.findElement(FAQ1).click(); }
```
**Можно лучше:** тут вполне можно было обойтись одной функцией, без копипасты

egorvas commented on src/test/java/pages/MainPageTest.java:
```diff
@@ -0,0 +1,97 @@
+    private WebDriver driver;
+
+    @Test
+    public void checkFAQ() {
```
**Нужно исправить:** такое решение не подходит. Тебе нужно проверять элементы FAQ независимо. Чтобы в случае если упадет тест по первому элементу FAQ, то остальные также бы проверились. Крайне рекомендую использовать для этого параметризацию, чтобы не дублировать код.

egorvas commented on src/test/java/pages/MainPageTest.java:
```diff
@@ -0,0 +1,97 @@
+    @Test
+    public void checkFAQ() {
+        // создали драйвер для браузера Chrome.
+        System.setProperty("webdriver.chrome.driver", "/Users/evgeniybatuev/Downloads/chromedriver_mac64/chromedriver");
```
**Нужно исправить:** нельзя хардкодить пути, у меня такого файла нет в системе, поэтому твои тесты не запускаются

egorvas commented on src/test/java/pages/MainPageTest.java:
```diff
@@ -0,0 +1,97 @@
+        // создали драйвер для браузера Chrome.
+        System.setProperty("webdriver.chrome.driver", "/Users/evgeniybatuev/Downloads/chromedriver_mac64/chromedriver");
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--remote-allow-origins=*");
```
**Нужно исправить:** все что не связано с логикой тестов нужно выносить в after/before функции

egorvas commented on src/test/java/pages/OrderPageTest.java:
```diff
@@ -0,0 +1,91 @@
+
+    private WebDriver driver;
+
+
```
**Можно лучше:** у тебя два фактически одинаковых теста, можно их объединить в один используя параметризацию.

Commit:
2 коммит

1. src/test/java/pages/MainPageTest.java
Нужно исправить: такое решение не подходит. Тебе нужно проверять элементы FAQ независимо. Чтобы в случае если упадет тест по первому элементу FAQ, то остальные также бы проверились. Крайне рекомендую использовать для этого параметризацию, чтобы не дублировать код.

!! Добавил MainPageParam и MainPageParamTest

3. src/test/java/pages/MainPageTest.java
Нужно исправить: нельзя хардкодить пути, у меня такого файла нет в системе, поэтому твои тесты не запускаются

!! Исправлено

4. src/test/java/pages/MainPageTest.java
Нужно исправить: все что не связано с логикой тестов нужно выносить в after/before функции

!! Исправлено

Commit:
2 коммит - Добавил задание из ФИНАЛЬНОГО задания по СЕЛЕНИУМУМУ (Metro)
